driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -171,15 +183,33 @@ public void run() {
   // following threshold, we just replace the connection by a new one.
   private final int minAllowedStreams;
 
-  HostConnectionPool(Host host, HostDistance hostDistance, SessionManager manager) {
+  HostConnectionPool(Host host, HostDistance hostDistance, final SessionManager manager) {
     assert hostDistance != HostDistance.IGNORED;
     this.host = host;
     this.hostDistance = hostDistance;
     this.manager = manager;
 
     this.minAllowedStreams = options().getMaxRequestsPerConnection(hostDistance) * 3 / 4;
 
-    this.timeoutsExecutor = manager.getCluster().manager.connectionFactory.eventLoopGroup.next();
+    this.scheduler = manager.getCluster().manager.connectionFactory.eventLoopGroup.next();
+
+    scheduler.schedule(
```

Comment:
Why do you need this rate-limiting? The driver opens at most connectionsPerShard connections to each shard at a time. Is it too many for you?